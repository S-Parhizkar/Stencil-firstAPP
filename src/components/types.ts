export interface ITodo {
    completed: boolean,
    id: number,
    order: number,
    title: string,
    url: string
} 
