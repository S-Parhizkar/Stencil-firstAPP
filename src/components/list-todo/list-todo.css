:host {
  display: block;
  height: 750px;
  overflow: auto;
}
