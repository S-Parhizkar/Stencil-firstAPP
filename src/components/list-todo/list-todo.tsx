import { Component, Host, h, Prop } from '@stencil/core';
import { ITodo } from '../types';
@Component({
  tag: 'list-todo',
  styleUrl: 'list-todo.css',
  shadow: true,
})
export class ListTodo {
  @Prop() todos: Array<ITodo> = [];

  render() {
    return (
      <Host>
        {this.todos.map(todo => (<to-do todo={todo} />
        ))}
      </Host>
    );
  }
}
