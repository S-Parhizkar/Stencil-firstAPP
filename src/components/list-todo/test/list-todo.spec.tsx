import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { ListTodo } from '../list-todo';
import { ITodo } from '../../types';


describe('list-todo', () => {
  const todo: ITodo = {
    completed: true,
    id: 2,
    order: 3,
    title: 'sport',
    url: 'https://my-list-to-do.netlify.app/',
  };

  let page: SpecPage;

  beforeEach(async ()=> {
    page = await newSpecPage({
      components: [ListTodo],
      template: () => <list-todo todo={todo}></list-todo>,
    });
    await page.waitForChanges();
  })

it('render', ()=>{
  expect(page.root).toMatchSnapshot();
})

});
