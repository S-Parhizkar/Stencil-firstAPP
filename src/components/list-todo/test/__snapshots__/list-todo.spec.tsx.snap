// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list-todo render 1`] = `
<list-todo>
  <mock:shadow-root></mock:shadow-root>
</list-todo>
`;
